### ISSUE TITLE
We need tag:yaml.org,2002:merge

### ISSUE THREAD

**Issue by menduz** [mulesoft-labs/yaml-ast-parser#3]
When we are trying to parse this valid YAML

``` yaml
object: &a
  a: 1

output:
  <<: *a
  b: 2
```

It fails with

```
Error: Should not happen
    at storeMappingPair (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:254:15)
    at readBlockMapping (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:889:17)
    at composeNode (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:1093:21)
    at readBlockMapping (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:880:17)
    at composeNode (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:1093:21)
    at readDocument (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:1236:5)
    at loadDocuments (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:1280:9)
    at Object.load (/Users/agustinmendez/Code/http-bat/node_modules/yaml-ast-parser/dist/loader.js:1301:21)

```

I see this fragment on the actual code:

``` typescript
...
  if ('tag:yaml.org,2002:merge' === keyTag) {
      throw new Error("Should not happen")//TODO LOOK IN IT
    //if (Array.isArray(valueNode)) {
    //  for (index = 0, quantity = valueNode.length; index < quantity; index += 1) {
    //    mergeMappings(state, _result, valueNode[index]);
    //  }
    //} else {
    //  mergeMappings(state, _result, valueNode);
    //}
  } else {
...
```

We need it not to break the code, it could actually not perform the merge because this is a parser and not an interprete. But the failure is blocking us!

**Comment by menduz** [comment]
@sichvoge

**Comment by ddenisenko** [comment]
It does not explode now, released in 0.0.29.
JS Parser supports it in develop branch, it will be published at Monday.
